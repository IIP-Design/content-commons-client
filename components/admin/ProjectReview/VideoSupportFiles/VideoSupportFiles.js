import React from 'react';
import PropTypes from 'prop-types';
import { gql } from '@apollo/client';
import { graphql } from '@apollo/client/react/hoc';
import orderBy from 'lodash/orderBy';
import { Icon, Loader } from 'semantic-ui-react';

import { getCount } from 'lib/utils';
import VideoSupportFilesItem from './VideoSupportFilesItem/VideoSupportFilesItem';

import './VideoSupportFiles.scss';

const VideoSupportFiles = props => {
  const { error, loading, project } = props.data;

  if ( loading ) {
    return (
      <div
        style={ {
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center',
          justifyContent: 'center',
          minHeight: '200px',
        } }
      >
        <Loader
          active
          inline="centered"
          style={ { marginBottom: '1em' } }
          content="Loading support file(s)..."
        />
      </div>
    );
  }

  if ( error ) {
    return (
      <div className="video-support-files error">
        <p>
          <Icon color="red" name="exclamation triangle" />
          <span>Loading error...</span>
        </p>
      </div>
    );
  }

  if ( !project || !Object.keys( project ).length ) return null;

  const { srts, additionalFiles, thumbnails } = project;

  const srtsCount = getCount( srts );
  const additionalFilesCount = getCount( additionalFiles );
  const thumbnailsCount = getCount( thumbnails );
  const totalFilesCount = srtsCount + additionalFilesCount + thumbnailsCount;

  const hasSRTs = srtsCount > 0;
  const hasAdditionalFiles = additionalFilesCount > 0;
  const hasThumbnails = thumbnailsCount > 0;
  const hasSupportFiles = totalFilesCount > 0;

  const additionalFilesSorted = hasAdditionalFiles
    ? orderBy( additionalFiles, ['filetype'] )
    : [];

  const allAdditionalFiles = hasThumbnails
    ? [...additionalFilesSorted, ...thumbnails]
    : [...additionalFilesSorted];

  if ( !hasSupportFiles ) return null;

  return (
    <section className="section section--project_support-files project_support-files">
      <h3 className="uppercase">
        { `Support File${totalFilesCount > 1 ? 's' : ''}` }
      </h3>

      { hasSRTs
        && (
          <section className="files section">
            <h4 id="srt-files">
              { `SRT file${srtsCount > 1 ? 's' : ''}` }
            </h4>
            <ul aria-describedby="srt-files">
              { srts.map( srt => (
                <VideoSupportFilesItem key={ srt.id } file={ srt } />
              ) ) }
            </ul>
          </section>
        ) }

      { ( hasAdditionalFiles || hasThumbnails )
        && (
          <section className="addtl_files section">
            <h4 id="additional-files">
              { `Additional file${additionalFilesCount + thumbnailsCount > 1 ? 's' : ''}` }
            </h4>
            <ul aria-describedby="additional-files">
              { allAdditionalFiles.map( file => (
                <VideoSupportFilesItem key={ file.id } file={ file } />
              ) ) }
            </ul>
          </section>
        ) }
    </section>
  );
};

VideoSupportFiles.propTypes = {
  id: PropTypes.string, // eslint-disable-line
  data: PropTypes.object,
};

const VIDEO_PROJECT_REVIEW_SUPPORT_FILES_QUERY = gql`
  query VideoProjectReviewSupportFiles($id: ID!) {
    project: videoProject(id: $id) {
      id
      srts: supportFiles(
        where: {
          OR: [
            { filename_ends_with: "srt" },
            { filetype_contains: "application/x-subrip" }
          ]
        },
        orderBy: filename_ASC
      ) {
        id
        filename
        language {
          id
          displayName
        }
      }
      additionalFiles: supportFiles(
        where: {
          OR: [
            { filename_not_ends_with: "srt" },
            { filetype_not_contains: "application/x-subrip" }
          ]
        },
        orderBy: filename_ASC
      ) {
        id
        filename
        filetype
        language {
          id
          displayName
        }
      }
      thumbnails( orderBy: filename_ASC ) {
        id
        filename
        language {
          id
          displayName
        }
      }
    }
  }
`;

export default graphql( VIDEO_PROJECT_REVIEW_SUPPORT_FILES_QUERY, {
  options: props => ( {
    variables: {
      id: props.id,
    },
    fetchPolicy: 'no-cache',
  } ),
} )( VideoSupportFiles );

export { VIDEO_PROJECT_REVIEW_SUPPORT_FILES_QUERY };
