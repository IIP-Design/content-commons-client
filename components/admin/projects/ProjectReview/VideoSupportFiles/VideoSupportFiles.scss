.project_support-files {
  box-shadow: none !important;

  p {
    line-height: 1.5285em; 
  }

  .thumbnails {

    &_img {
      width: 100%;
    }        
  }

  .checkbox {
    margin-top: 1em;
  }
}