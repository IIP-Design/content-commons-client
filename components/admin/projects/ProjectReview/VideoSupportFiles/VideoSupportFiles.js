/**
 *
 * VideoSupportFiles
 *
 */
import React, { useEffect, useState } from 'react';
import { func, object, string } from 'prop-types';
import gql from 'graphql-tag';
import { compose, graphql } from 'react-apollo';
import orderBy from 'lodash/orderBy';
import { Checkbox, Loader } from 'semantic-ui-react';

import { getPluralStringOrNot } from 'lib/utils';

import './VideoSupportFiles.scss';

const VideoSupportFiles = props => {
  const { error, loading, project } = props.data;

  const [protectImages, setProtectImages] = useState( project.protectImages );

  const updateState = () => setProtectImages( project.protectImages );

  useEffect( updateState );

  if ( loading ) {
    return (
      <div style={ {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center',
        minHeight: '200px'
      } }
      >
        <Loader
          active
          inline="centered"
          style={ { marginBottom: '1em' } }
          content="Loading support file(s)..."
        />
      </div>
    );
  }

  if ( error ) return `Error! ${error.message}`;

  if ( !project || !Object.keys( project ).length ) return null;

  const handleUpdateProtectImages = () => {
    const { id, updateProtectImages } = props;
    updateProtectImages( {
      variables: {
        data: { protectImages: !protectImages },
        where: { id }
      }
    } );
  };

  const { srts, additionalFiles } = project;
  const additionalFilesSorted = orderBy( additionalFiles, ['filetype'] );

  return (
    <section className="section section--project_support-files project_support-files">
      <h3 className="uppercase">
        { getPluralStringOrNot( [...srts, ...additionalFiles], 'Support File' ) }
      </h3>

      <section className="files section">
        <h4 id="srt-files">
          { getPluralStringOrNot( srts, 'SRT file' ) }
        </h4>
        <ul aria-describedby="srt-files">
          { srts.map( srt => (
            <li key={ srt.id } className="file">
              <b className="label">{ srt.language.displayName }:</b> { srt.filename }
            </li>
          ) ) }
        </ul>
      </section>

      <section className="addtl_files section">
        <h4 id="additional-files">
          { `Additional file${additionalFilesSorted.length > 1 ? 's' : ''}` }
        </h4>
        <ul aria-describedby="additional-files">
          { additionalFilesSorted.map( file => (
            <li key={ file.id }><b className="label">{ file.language.displayName }:</b> { file.filename }</li>
          ) ) }
        </ul>
      </section>

      <Checkbox
        label="Disable right-click to protect your images"
        checked={ !!protectImages }
        onClick={ handleUpdateProtectImages }
      />
    </section>
  );
};

VideoSupportFiles.propTypes = {
  id: string,
  data: object,
  updateProtectImages: func
};

const VIDEO_PROJECT_REVIEW_SUPPORT_FILES_QUERY = gql`
  query VideoProjectReviewSupportFiles($id: ID!) {
    project: videoProject(id: $id) {
      id
      protectImages
      srts: supportFiles(
        where: {filetype: "srt"},
        orderBy: filename_ASC
      ) {
        id
        filename
        language {
          displayName
        }
      }
      additionalFiles: supportFiles(
        where: {filetype_not: "srt"},
        orderBy: filename_ASC
      ) {
        id
        filename
        filetype
        language {
          displayName
        }
      }
    }
  }
`;

const UPDATE_PROTECT_IMAGES_MUTATION = gql`
  mutation UpdateProtectImages($data: VideoProjectUpdateInput!
  $where: VideoProjectWhereUniqueInput!) {
    updateVideoProject(data: $data, where: $where) {
      id
      protectImages
    }
  }
`;

const supportFilesQuery = graphql( VIDEO_PROJECT_REVIEW_SUPPORT_FILES_QUERY, {
  options: props => ( {
    variables: {
      id: props.id
    },
  } )
} );

const updateProtectImagesMutation = graphql(
  UPDATE_PROTECT_IMAGES_MUTATION,
  { name: 'updateProtectImages' }
);

export default compose(
  updateProtectImagesMutation,
  supportFilesQuery
)( VideoSupportFiles );

export {
  VIDEO_PROJECT_REVIEW_SUPPORT_FILES_QUERY,
  UPDATE_PROTECT_IMAGES_MUTATION
};
