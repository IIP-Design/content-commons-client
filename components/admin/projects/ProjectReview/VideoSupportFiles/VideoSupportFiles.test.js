import React from 'react';
import { shallow } from 'enzyme';

import VideoSupportFiles from './VideoSupportFiles';

describe( '<VideoSupportFiles />', () => {
  it( 'renders without crashing', () => {
    shallow( <VideoSupportFiles /> );
  } );
} );
