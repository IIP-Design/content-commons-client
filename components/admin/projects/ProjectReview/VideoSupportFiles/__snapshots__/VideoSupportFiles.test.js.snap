// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoSupportFiles /> renders initial loading state without crashing 1`] = `
<VideoSupportFiles
  data={
    Object {
      "error": undefined,
      "fetchMore": [Function],
      "loading": true,
      "networkStatus": 1,
      "refetch": [Function],
      "startPolling": [Function],
      "stopPolling": [Function],
      "subscribeToMore": [Function],
      "updateQuery": [Function],
      "variables": Object {
        "id": "123",
      },
    }
  }
  id="123"
  updateProtectImages={[Function]}
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "minHeight": "200px",
      }
    }
  >
    <Loader
      active={true}
      content="Loading support file(s)..."
      inline="centered"
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    >
      <div
        className="ui active text centered inline loader"
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Loading support file(s)...
      </div>
    </Loader>
  </div>
</VideoSupportFiles>
`;

exports[`<VideoSupportFiles /> renders the final state 1`] = `
<VideoSupportFiles
  data={
    Object {
      "error": undefined,
      "fetchMore": [Function],
      "loading": false,
      "networkStatus": 7,
      "project": Object {
        "additionalFiles": Array [
          Object {
            "filename": "image-1.jpg",
            "filetype": "jpg",
            "id": "82kw",
            "language": Object {
              "displayName": "English",
            },
          },
          Object {
            "filename": "image-2.jpg",
            "filetype": "jpg",
            "id": "28zi",
            "language": Object {
              "displayName": "French",
            },
          },
        ],
        "id": "123",
        "protectImages": null,
        "srts": Array [
          Object {
            "filename": "srt-1.srt",
            "id": "34is",
            "language": Object {
              "displayName": "English",
            },
          },
          Object {
            "filename": "srt-2.srt",
            "id": "48sa",
            "language": Object {
              "displayName": "French",
            },
          },
        ],
      },
      "refetch": [Function],
      "startPolling": [Function],
      "stopPolling": [Function],
      "subscribeToMore": [Function],
      "updateQuery": [Function],
      "variables": Object {
        "id": "123",
      },
    }
  }
  id="123"
  updateProtectImages={[Function]}
>
  <section
    className="section section--project_support-files project_support-files"
  >
    <h3
      className="uppercase"
    >
      Support Files
    </h3>
    <section
      className="files section"
    >
      <h4
        id="srt-files"
      >
        SRT files
      </h4>
      <ul
        aria-describedby="srt-files"
      >
        <li
          className="file"
          key="34is"
        >
          <b
            className="label"
          >
            English
            :
          </b>
           
          srt-1.srt
        </li>
        <li
          className="file"
          key="48sa"
        >
          <b
            className="label"
          >
            French
            :
          </b>
           
          srt-2.srt
        </li>
      </ul>
    </section>
    <section
      className="addtl_files section"
    >
      <h4
        id="additional-files"
      >
        Additional files
      </h4>
      <ul
        aria-describedby="additional-files"
      >
        <li
          key="82kw"
        >
          <b
            className="label"
          >
            English
            :
          </b>
           
          image-1.jpg
        </li>
        <li
          key="28zi"
        >
          <b
            className="label"
          >
            French
            :
          </b>
           
          image-2.jpg
        </li>
      </ul>
    </section>
    <Checkbox
      checked={false}
      label="Disable right-click to protect your images"
      onClick={[Function]}
      type="checkbox"
    >
      <div
        className="ui checkbox"
        onChange={[Function]}
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <input
          checked={false}
          className="hidden"
          readOnly={true}
          tabIndex={0}
          type="checkbox"
        />
        <label>
          Disable right-click to protect your images
        </label>
      </div>
    </Checkbox>
  </section>
</VideoSupportFiles>
`;
